Octokit/Http/Connection.cs

```diff
@@ -136,6 +136,30 @@ public Connection(ProductHeaderValue productInformation, Uri baseAddress, ICrede
             _jsonPipeline = new JsonHttpPipeline();
         }
 
+        /// <summary>
+        /// Gets the latest API Info - this will be null if no API calls have been made
+        /// </summary>
+        /// <returns><seealso cref="ApiInfo"/> representing the information returned as part of an Api call</returns>
+        public ApiInfo LastApiInfo
+        { 
+            get
+            {
+                lock (LastApiInfoLocker)
+                {
+                    return _lastApiInfo;
+                }
+            }
+            private set
+            {
+                lock (LastApiInfoLocker)
+                {
+                    _lastApiInfo = value;
+                }
+            }
+        }
+        private ApiInfo _lastApiInfo;
+        private readonly object LastApiInfoLocker = new object();
+
```

Comment:
Hmm, do we actually need locks here? If someone is reading the value _just_ as another thread is writing the value, it sort of doesn't matter who wins. After all, the read value will only be _slightly_ outdated.

In other words, I don't see any problems with race conditions in this case as this isn't data that's timing sensitive. I don't think torn reads are a problem because these are 32 bit references.

This value is going to be set on every request, I'd rather avoid a lock here unless it's absolutely needed.